tutorials/LinearAlgebra/LinearAlgebra.ipynb

```diff
@@ -0,0 +1,983 @@
+    "\n",
+    "$$\\begin{bmatrix}\n",
+    "    x_{0,0} & x_{0,1} & \\dotsb & x_{0,m-1} \\\\\n",
+    "    x_{1,0} & x_{1,1} & \\dotsb & x_{1,m-1} \\\\\n",
+    "    \\vdots  & \\vdots  & \\ddots & \\vdots  \\\\\n",
+    "    x_{n-1,0} & x_{n-1,1} & \\dotsb & x_{n-1,m-1}\n",
+    "\\end{bmatrix}\n",
+    "+\n",
+    "\\begin{bmatrix}\n",
+    "    y_{0,0} & y_{0,1} & \\dotsb & y_{0,m-1} \\\\\n",
+    "    y_{1,0} & y_{1,1} & \\dotsb & y_{1,m-1} \\\\\n",
+    "    \\vdots  & \\vdots  & \\ddots & \\vdots  \\\\\n",
+    "    y_{n-1,0} & y_{n-1,1} & \\dotsb & y_{n-1,m-1}\n",
+    "\\end{bmatrix}\n",
+    "=\n",
+    "\\begin{bmatrix}\n",
+    "    x_{0,0} + y_{0,0} & x_{0,1} + y_{0,1} & \\dotsb & x_{0,m-1} + y_{0,m-1} \\\\\n",
+    "    x_{1,0} + y_{1,0} & x_{1,1} + y_{1,1} & \\dotsb & x_{1,m-1} + y_{1,m-1} \\\\\n",
+    "    \\vdots  & \\vdots  & \\ddots & \\vdots  \\\\\n",
+    "    x_{n-1,0} + y_{n-1,0} & x_{n-1,1} + y_{n-1,1} & \\dotsb & x_{n-1,m-1} + y_{n-1,m-1}\n",
+    "\\end{bmatrix}$$\n",
+    "\n",
+    "Matrix addition has the following properties:\n",
+    "\n",
+    "* Commutativity: $A + B = B + A$\n",
+    "* Associativity: $(A + B) + C = A + (B + C)$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### <span style=\"color:blue\">Exercise 1</span>: Matrix addition.\n",
+    "\n",
+    "**Inputs:**\n",
+    "\n",
+    "1. An $n \\times m$ matrix $A$, represented as a two-dimensional list.\n",
+    "2. An $n \\times m$ matrix $B$, represented as a two-dimensional list.\n",
+    "\n",
+    "**Goal:** Return the matrix resulting from $A + B$, represented as a two-dimensional list.\n",
+    "\n",
+    "> When representing matrices as lists, each sub-list represents a row.\n",
+    ">\n",
+    "> For example, list `[[1, 2], [3, 4]]` represents the following matrix:\n",
+    ">\n",
+    "> $$\\begin{bmatrix}\n",
+    "    1 & 2 \\\\\n",
+    "    3 & 4\n",
+    "\\end{bmatrix}$$\n",
+    "\n",
+    "Fill in the missing code and run the cell below to test your work."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@exercise\n",
+    "def matrix_add(a : Matrix, b : Matrix) -> Matrix:\n",
+    "    # You can get the size of a matrix like so:\n",
+    "    n = len(a)\n",
+    "    m = len(a[0])\n",
+    "    \n",
+    "    # You can use this function to initialize a matrix filled with 0s\n",
+    "    c = create_empty_matrix(n, m)\n",
+    "    \n",
+    "    # You can use a for loop to do something n (or m) times\n",
+    "    for i in range(n):\n",
+    "        # Loops can be nested\n",
+    "        for j in range(m):\n",
+    "            # You can access elements of a matrix like this\n",
+    "            x = a[i][j]\n",
```

Comment:
Following the approach we used in CompexArithmetic, I would add a line `y = b[i][j]` here, so that the learner has to modify just the line marked with `...`